astropy/stats/bayesian_blocks.py

```diff
@@ -242,7 +238,8 @@ def validate_input(self, t, x=None, sigma=None):
         # if x is specified, then we need to simultaneously sort t and x
         else:
             # TODO: allow broadcasted x?
-            x = np.asarray(x)
+            x = np.asarray(x, dtype=float)
```

Comment:
🤷‍♀ whichever is preferred, I hasn't aware of opinions on this. I'll change it to np.float once I get the review from @astrofrog and/or @larrybradley.

Reply:
ok, since there is no more input on this, and this is just to use the same approach as it was in here originally (see few lines above for `t`), I would suggest to go ahead and merge. We have 79 other cases of `dtype=float`, so if that's an issue, it may need a separate PR anyway.

Reply:
Fine by me!

Reply:
My advise was/is never to use `np.float`. It's just `float` anyway. I prefer `float` or if one wants to me more precise `np.float64` (same result).

Reply:
OK, that indeed makes sense, and there is only 2 cases of that...

Reply:
> Care to open a PR for it, or shall I?

I can do that. I'll also dig up the original clean-up issue/pr. :)

Reply:
The PR where I originally discussed and removed them was #6603 and the PR that does the same again is #9526 .

Also sorry that I responded so late. This PR somehow slipped by.

And I forgot to mention in my previous commit that `np.float_` is also acceptable (which is the dtype corresponding to `float` and it's alias `np.float`).